network/p2p/router.go

```diff
@@ -128,14 +128,15 @@ func (r *router) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID ui
 	start := time.Now()
 	parsedMsg, handler, handlerID, ok := r.parse(request)
 	if !ok {
-		r.log.Debug("failed to process message",
+		r.log.Debug("received message for unregistered handler",
```

Comment:
On a quick look - this is a bit confusing. It feels like failing to parse a message is separable from using an unregistered handler...

On closer inspection `parse` is really just parsing the handlerID... Is it worth commenting that providing an invalid handlerID (aka, empty bytes or overflowing `uint64`) is treated the same as specifying a handler that wasn't registered?

Reply:
Yeah maybe we could rename this function... I can leave a comment on this case.